File: 13-roman-to-integer/13-roman-to-integer.java
class Solution {
    public int romanToInt(String s) {
         HashMap<Character , Integer> map = new HashMap<>();
        map.put('I' , 1);
        map.put('V' , 5);
        map.put('X' , 10);
        map.put('L' , 50);
        map.put('C' , 100);
        map.put('D' , 500);
        map.put('M' , 1000);
        
        int sum = 0 ;
        
        int index = 0;
        while ( index < s.length())
        {
            int cur = map.get(s.charAt(index));
            index++;
            if (index  < s.length())
            {
                int cur2 = map.get(s.charAt(index));
                if (cur2 > cur){
                    sum+=(cur2 - cur);
                    index++;
                    }
                else
                    sum+=cur;
            }else
                sum+=cur;
            
                
        }
        return sum;
    }
}

File: 13-roman-to-integer/13-roman-to-integer.cpp
class Solution {
public:
    int romanToInt(string s) {
        map<char, int> mp;
        mp['I'] = 1, mp['V'] = 5, mp['X'] = 10, mp['L'] = 50, mp['C'] = 100, mp['D'] = 500, mp['M'] = 1000;
        int n = s.length();
        int sum=0;
        sum = mp[s[0]];
        for(int i=1; i<n; i++){
            if(mp[s[i]] > mp[s[i-1]])
                // sum += mp[s[i]];
                sum = sum + mp[s[i]] - 2*mp[s[i-1]];
            else
                sum += mp[s[i]];
        }
        return sum;
    }
};